Sources/ProtobufferCoding/Proto/Descriptor.swift

```diff
@@ -0,0 +1,1115 @@
+        self.name = name
+        self.package = package
+        self.dependencies = dependencies
+        self.publicDependency = publicDependency
+        self.weakDependency = weakDependency
+        self.messageTypes = messageTypes
+        self.enumTypes = enumTypes
+        self.services = services
+        self.extensions = extensions
+        self.options = options
+        self.sourceCodeInfo = sourceCodeInfo
+        self.syntax = syntax
+    }
+}
+
+
+/// Describes a message type.
+public struct DescriptorProto: Codable, Hashable, ProtobufMessageWithCustomFieldMapping, _ProtoPackage_Google_Protobuf {
+    public var name: String
+    
+    public let fields: [FieldDescriptorProto]
+    public let extensions: [FieldDescriptorProto]
+    
+    public var nestedTypes: [DescriptorProto]
+    public var enumTypes: [EnumDescriptorProto]
+    
+    public struct ExtensionRange: Codable, ProtobufMessage, Hashable, _ProtoPackage_Google_Protobuf {
+        public let start: Int32?
+        public let end: Int32?
+        public let options: ExtensionRangeOptions?
+    }
+    public let extensionRanges: [ExtensionRange]
+    public let oneofDecls: [OneofDescriptorProto]
+    public let options: MessageOptions?
+    
+    
+    /// Range of reserved tag numbers. Reserved tag numbers may not be used by
+    /// fields or extension ranges in the same message. Reserved ranges may
+    /// not overlap.
+    public struct ReservedRange: Codable, ProtobufMessage, Hashable, _ProtoPackage_Google_Protobuf {
+        /// Inclusive.
+        public let start: Int32?
+        /// Exclusive.
+        public let end: Int32?
+    }
+    public let reservedRanges: [ReservedRange]
+    
+    /// Reserved field names, which may not be used by fields in the same message.
+    /// A given name may only be reserved once.
+    public let reservedNames: [String]
+    
+    
+    public enum CodingKeys: Int, ProtobufMessageCodingKeys {
+        case name = 1
+        case fields = 2
+        case extensions = 6
+        case nestedTypes = 3
+        case enumTypes = 4
+        case extensionRanges = 5
+        case oneofDecls = 8
+        case options = 7
+        case reservedRanges = 9
+        case reservedNames = 10
+    }
+}
+
+
+public struct ExtensionRangeOptions: Codable, Hashable, ProtobufMessageWithCustomFieldMapping, _ProtoPackage_Google_Protobuf {
+    /// The parser stores options it doesn't recognize here. See above.
+//  repeated UninterpretedOption uninterpreted_option = 999;
+    public let uninterpretedOptions: [UninterpretedOption]
+
+
+//  // Clients can define custom options in extensions of this message. See above.
+//  extensions 1000 to max;
+    
```

Comment:
The entire Descriptors file was essentially just copied from [the source](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto), and i then commented out the proto code and added matching Swift code directly below. So the commented parts aren't really documentation, but rather the corresponding proto code. Since not all proto features are supported by us atm (extensions being one of the missing pieces), I deciced to keep them in anyway, to make it directly visible that there's missing stuff

Reply:
The proto file was turned to swift code by hand? Is there a particular reason why we didn't just use code generation for that? Seems a bit more sensible for future maintenance then comparing changes by hand.
Otherwise, the `swift-protobuf` has the same file generated in their project: [descriptor.pb.swift](https://github.com/apple/swift-protobuf/blob/cf07a8889859ef191721094fa52713c834cf1481/Sources/SwiftProtobuf/descriptor.pb.swift) or their API wrapper around it [Descirptor.swift](https://github.com/apple/swift-protobuf/blob/cf07a8889859ef191721094fa52713c834cf1481/Sources/SwiftProtobufPluginLibrary/Descriptor.swift). Not sure if we want to pull in that dependency just for that file.

Reply:
It's a one-off, so it doesn't really make sense to automate it. Sure it might be changed at some point in the future, but if that were to happen a) it wouldn't be too problematic since the proto format is backward-compatible enough that our definitions would still be compatible, and b) the only way i'd see them changing the descriptors types would be as part of e.g. a proto4 spec, which would then probably break our coding implementation anyway.

Additionally, swift-protobuf would've turned it into Swift code compatible with their interfaces and protocols, whereas we need it to be compatible with the protocols used by ProtobufferCoding. It wouldn't be possible to simply import and use the proto definition types from swift-protobuf, since they're using an entirely different set of protocols and conventions to do their proto message type <-> swift type definition mapping.